app/models/user_authenticate.rb

```diff
@@ -0,0 +1,44 @@
+class UserAuthenticate
+  def initialize(auth_hash)
+    @auth_subject_id = auth_hash.uid
+    @auth_expires_at = Time.zone.now + auth_hash.credentials.expires_in
+    @last_auth_at = Time.zone.now
+    @auth_info = auth_hash.info
+  end
```

Comment:
Does the `auth_hash` need any fields customisation or are `credentials`, `uid` etc the standard response values?

Reply:
There's an `auth_details ` further below so would be good to differentiate where `auth_hash` comes from versus `auth_details` - may just be a naming thing.

Reply:
Is there a context in which `auth_hash` is nil or has missing `info` for example? E.g. because of some setting in AD?

Reply:
@mseedat-moj  uid is standard, and in this case comes from the verified id_token.
- good point on auth details, will clarify.
- a missing auth hash is handled at the rack layer by the warden strategy here https://github.com/ministryofjustice/laa-review-criminal-legal-aid/blob/main/lib/warden/strategies/azure_ad.rb#L10